src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -30,133 +24,17 @@ public class EmbedAppNameInHost : TaskBase
-            using (FileStream fs = new FileStream(ModifiedAppHostPath, FileMode.Truncate, FileAccess.ReadWrite, FileShare.Read))
-            {
-                fs.Write(array, 0, array.Length);
-            }
-        }
-
-        // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
-        private static int[] ComputeKMPFailureFunction(byte[] pattern)
-        {
-            int[] table = new int[pattern.Length];
-            if (pattern.Length >= 1)
-            {
-                table[0] = -1;
-            }
-            if (pattern.Length >= 2)
-            {
-                table[1] = 0;
-            }
-
-            int pos = 2;
-            int cnd = 0;
-            while (pos < pattern.Length)
-            {
-                if (pattern[pos - 1] == pattern[cnd])
-                {
-                    table[pos] = cnd + 1;
-                    cnd++;
-                    pos++;
-                }
-                else if (cnd > 0)
-                {
-                    cnd = table[cnd];
-                }
-                else
-                {
-                    table[pos] = 0;
-                    pos++;
-                }
-            }
-            return table;
-        }
-
-        // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
-        private static int KMPSearch(byte[] pattern, byte[] bytes)
-        {
-            int m = 0;
-            int i = 0;
-            int[] table = ComputeKMPFailureFunction(pattern);
-
-            while (m + i < bytes.Length)
-            {
-                if (pattern[i] == bytes[m + i])
-                {
-                    if (i == pattern.Length - 1)
-                    {
-                        return m;
-                    }
-                    i++;
-                }
-                else
-                {
-                    if (table[i] > -1)
-                    {
-                        m = m + i - table[i];
-                        i = table[i];
-                    }
-                    else
-                    {
-                        m++;
-                        i = 0;
-                    }
-                }
-            }
-            return -1;
-        }
-
-        private static  void SearchAndReplace(byte[] array, byte[] searchPattern, byte[] patternToReplace, string appHostSourcePath)
-        {
-            int offset = KMPSearch(searchPattern, array);
-            if (offset < 0)
-            {
-                throw new BuildErrorException(Strings.AppHostHasBeenModified, appHostSourcePath, placeHolder);
-            }
-
-            patternToReplace.CopyTo(array, offset);
-
-            if (patternToReplace.Length < searchPattern.Length)
-            {
-                for (int i = patternToReplace.Length; i < searchPattern.Length; i++)
-                {
-                    array[i + offset] = 0x0;
-                }
-            }
+            EmbedAppNameInHostUtil.EmbedAppHost(
```

Comment:
I'm allergic to `Util` suffix. :) We have Utilities in a few places, but even that bugs me. It leads to grab bags of stuff. Also, EmbedAppHost reads like we're embedding the app host in something, but we're actually embedding something in the app host.

I think this could be something like `AppHost.Create`. Still thinking about the name. What do others think? cc @dsplaisted @peterhuene

Reply:
I had similar comments when we copied this into CLI. I understand the desire to link it back to `EmbedAppNameInHost`, but that's a verb because it's an MSBuild task; it's not a good name for something responsible for creating new app hosts.

Personally, I'm 👍 with `AppHost.Create`.

Reply:
I don't like Util as well but I need to change something to avoid name conflict with the task. AppHost.Create sounds good.

Reply:
Done